module6/ex28_tamperserve/ex28_tamperserve.py

```diff
@@ -0,0 +1,95 @@
+import hashlib
+import os
+import pickle
+import time
+
+from flask import Flask, request
+
+
+def get_sha1(pathfile):
+    with open(pathfile, 'rb') as f:
+        read_content = f.read()
+    file_hashed_sha1 = hashlib.sha1(read_content).hexdigest()
+    return file_hashed_sha1
+
+
+class FileInfo:
+    def __init__(self, filename, mtime, sha1):
+        self.filename = filename
+        self.mtime = mtime
+        self.sha1 = sha1
+
+    def __str__(self):
+        return f'FileInfo for {self.filename}, mtime {self.mtime}, sha1 {self.sha1}'
+
+    def __eq__(self, other):
+        return type(other) == type(self) and self.filename == other.filename
+
+
+class FileList:
+    def __init__(self, dir_path):
+        self.dir_path = dir_path
+        self.all_file_infos = []
+        self.last_timestamp = 0
+
+
+app = Flask(__name__)
+
+
+@app.route("/scan")
+def scan():
+    dir_path = request.args['dirpath']
+    if not os.path.isdir(dir_path):
+        return f'The given directory is not a valid one.'
+    else:
+        fl = FileList(dir_path)
+        for (root, dirs, files) in os.walk(dir_path, topdown=True):
+            for file in files:
+                pathfile = os.path.join(root, file)
+                file_info = os.stat(pathfile)
+                fl.all_file_infos.append(FileInfo(file, file_info.st_mtime, get_sha1(pathfile)))
+        fl.last_timestamp = time.time()
+        with open(dir_path.replace('/', '-'), "wb") as f:
+            pickle.dump(fl, f)
+        return "scan"
+
+
+@app.route("/rescan")
+def rescan():
+    dir_path = request.args['dirpath']
+    if not os.path.isdir(dir_path):
+        return f'The given directory is not a valid one.'
+    else:
+        is_scanned = False
+        pickle_name_dir = dir_path.replace('/', '-')
```

Comment:
After obtaining pickle file you should check if it exists on the disk, i.e. if it was previously scanned (pickle saved on line 52), if not return a corresponding message.

I see there's some logic regarding this in the next lines of code, but i think that's not correct. A `FileList` object is pickled and pickle file name is the same as the directory passed to `FileList` object.

Reply:
I think it's not the same.
  For directory _"/Users/raluca.romascu/python_upskilling_program/module6/ex28_tamperserve"_,
  name for pickled file looks like:
   _"-Users-raluca.romascu-python_upskilling_program-module6-ex28_tamperserve"_
 and I saved it in the corresponding directory. 

So to check if the folder has already been saved on the disk, I check if the file with the name specified above is found among the files in the given directory.